pymeasure/instruments/keysight/keysight34980A.py

```diff
@@ -0,0 +1,628 @@
+
+    def measure_frequency(self, chan, resolution = "DEFault", range_param = "DEFault"):    
+        """
+        The following command configures the instrument for frequency measurements, 
+        triggers the internal DMM to scan the specified channel.
+        """
+        temp_values = self.values(":MEASure:SCALar:FREQuency?")
+        return(str(temp_values[0]))
+
+    def measure_period(self, chan, resolution = "DEFault", range_param = "DEFault"):    
+        """
+        The following command configures the instrument for period measurements, 
+        triggers the internal DMM to scan the specified channel.    
+        """
+        temp_values = self.values(":MEASure:SCALar:PERiod?")
+        return(str(temp_values[0]))
+
+    def measure_resistance(self, chan, resolution = "DEFault", range_param = "AUTO"):    
+        """
+        The following command configures the instrument for resistance measurements, 
+        triggers the internal DMM to scan the specified channel.
+        """
+        temp_values = self.values(":MEASure:SCALar:RESistance?")
+        return(str(temp_values[0]))
+
+    def measure_temperature(self, chan, probe = "TCouple", probe_type = "T", resolution = "DEFault", range_param = "AUTO"):   
+        """
+        The following command configures the instrument for temperature measurements on the specified channel.
+        """
+        temp_values = self.values(":MEASure:SCALar:TEMPerature?")
+        return(str(temp_values[0]))
+
+    def set_temperature_unit(self, chan = "", unit = 'K'):    
+        """
+        This command selects the temperature units (°C, °F, or Kelvins) on the specified channels. 
+        If you omit the optional <ch_list> parameter, this command applies to the internal DMM, 
+        independent of any channels or a scan list.
+        """
+        if (chan == ""):
+            self.write(":UNIT:TEMPerature " + str(unit))
+            while((self.is_temperature_unit()) != str(str(unit) + "\n")):
+                self.write(":UNIT:TEMPerature " + str(unit))    
+        else:
+            self.write(":UNIT:TEMPerature " + str(unit) + ", (@" + str(chan) + ")")
+            while((self.is_temperature_unit(chan)) != str(str(unit) + "\n")):
+                self.write(":UNIT:TEMPerature " + str(unit) + ", (@" + str(chan) + ")")
+
+    def is_temperature_unit(self, chan = ""):    
+        """
+        The command checks the temperature units (°C, °F, or Kelvins) on the specified channels. 
+        If you omit the optional <ch_list> parameter, this command applies to the internal DMM, 
+        independent of any channels or a scan list. 
+        """
+        if (chan == ""):
+            self.write(":UNIT:TEMPerature?") 
+            temp_values = self.read()
+            return(str(temp_values))
+        else:
+            self.write(":UNIT:TEMPerature? (@" + str(chan) + ")") 
+            temp_values = self.read()
+            return(str(temp_values))
```

Comment:
The fact that the instrument has an internal DMM or a channel for a lot of commands complicates the implementation quite a bit; I'm wondering if there would be a neater way to implement this.

You could consider defining an `Instrument.control` property for the temperature unit of the DMM and have a sub-class for the different channels such that you can call Keysight34980A34980A.ch1.temperature_unit (see for example the KeysightDSOX1102G class where there are 2 channels).

If there are really many channels, you could also consider another construction where you have a sort of channelmanager that manages the channels such that in the end you can call it like Keysight34980A34980A.ch(1).temperature_unit:
```
class Keysight34980A34980A:
    def __init__(self):
        self.ch = ChannelManager()


class ChannelManager:
    def __init__(self):
        self.channel = 0

    def __call__(self, channel):
        self.channel = channel
        return self

    def do_something(self):
        print(f"doing something to channel {self.channel}")


a = Keysight34980A34980A()
a.ch(1).do_something()
a.ch(2).do_something()

# Calling without the number will use the last entered channel number
a.ch.do_something()

# You can't just store a channel, because you constantly refer to the same object
ch1 = a.ch(1)
ch2 = a.ch(2)
ch1.do_something()
# but you can using copy if you'd want
import copy
ch1 = copy.copy(a.ch(1))
ch2 = copy.copy(a.ch(2))
ch1.do_something()
ch2.do_something()
```